pyccl/tests/test_profiles.py

```diff
@@ -120,6 +120,42 @@ def test_hod_smoke():
         assert getattr(p, n) == 1234.
 
 
+@pytest.mark.parametrize('real_prof', [True, False])
+def test_hod_ns_independent(real_prof):
+    def func(prof):
+        return prof._real if real_prof else prof._fourier
+
+    c = ccl.halos.ConcentrationDuffy08(M200)
+    hmd = c.mdef
+    p1 = ccl.halos.HaloProfileHOD(c_M_relation=c,
+                                  lMmin_0=12.,
+                                  ns_independent=False)
+    p2 = ccl.halos.HaloProfileHOD(c_M_relation=c,
+                                  lMmin_0=12.,
+                                  ns_independent=True)
+    # M < Mmin
```

Comment:
Very nice tests!

Reply:
I've already done the test (for M < Mmin). Do you think it would make sense to include the other 2 cases (as with the `_real` and `_fourier` methods)?

Reply:
Hadn't seen those. OK, this looks good to me!